### ISSUE TITLE
You are advised to optimize format performance

### ISSUE THREAD

**Issue by zhuxiujia** [pydantic/speedate#16]
You are advised to optimize format performance
this change Can improve format performance several times

for example: this code
```
write!(f, "{:04}-{:02}-{:02}", self.year, self.month, self.day)
```
to My [Code](https://github.com/rbatis/fast_log/blob/master/src/date.rs)

```
impl Display for LogDate {
    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
       let mut buf: [u8; 29] = *b"0000-00-00 00:00:00.000000000";

        buf[0] = b'0' + (self.year / 1000) as u8;
        buf[1] = b'0' + (self.year / 100 % 10) as u8;
        buf[2] = b'0' + (self.year / 10 % 10) as u8;
        buf[3] = b'0' + (self.year % 10) as u8;


        buf[5] = b'0' + (self.mon / 10) as u8;
        buf[6] = b'0' + (self.mon % 10) as u8;

        buf[8] = b'0' + (self.day / 10) as u8;
        buf[9] = b'0' + (self.day % 10) as u8;

        buf[11] = b'0' + (self.hour / 10) as u8;
        buf[12] = b'0' + (self.hour % 10) as u8;
        buf[14] = b'0' + (self.min / 10) as u8;
        buf[15] = b'0' + (self.min % 10) as u8;
        buf[17] = b'0' + (self.sec / 10) as u8;
        buf[18] = b'0' + (self.sec % 10) as u8;

        buf[19] = b'.';

        buf[20] = b'0' + (self.nano / 100000000) as u8;
        buf[21] = b'0' + (self.nano / 10000000 % 10) as u8;
        buf[22] = b'0' + (self.nano / 1000000 % 10) as u8;
        buf[23] = b'0' + (self.nano / 100000 % 10) as u8;
        buf[24] = b'0' + (self.nano / 10000 % 10) as u8;
        buf[25] = b'0' + (self.nano / 1000 % 10) as u8;
        buf[26] = b'0' + (self.nano / 100 % 10) as u8;
        buf[27] = b'0' + (self.nano / 10 % 10) as u8;
        buf[28] = b'0' + (self.nano % 10) as u8;

        f.write_str(std::str::from_utf8(&buf[..]).unwrap())
    }
}
```

**Comment by samuelcolvin** [comment]
Humm, but how slow is `Display` now?

If it's from 2ns to 0.1ns, I really don't think it's worth it.

also, could some of this be moved into a look to save somewhat on verbosity?

PR welcome and  I'll review from there.

**Comment by zhuxiujia** [comment]
> Humm, but how slow is `Display` now?
> 
> If it's from 2ns to 0.1ns, I really don't think it's worth it.
> 
> also, could some of this be moved into a look to save somewhat on verbosity?
> 
> PR welcome and I'll review from there.

this befor and after benchmark
![微信图片_20220710184428](https://user-images.githubusercontent.com/3481148/178141591-9455c7a5-e789-4c55-b047-b960ac20d1c5.png)
![微信图片_20220710184437](https://user-images.githubusercontent.com/3481148/178141593-5b25ab6d-3c47-4832-b1f2-98687bb0f72f.png)

**Comment by samuelcolvin** [comment]
ok, PR welcome.

**Comment by samuelcolvin** [comment]
fixed by #17.
